packages/strapi-plugin-upload/admin/src/containers/InputModalStepperProvider/reducer.js

```diff
@@ -0,0 +1,169 @@
+import produce from 'immer';
```

Comment:
What was the issue with immutable I am ok to try using immer but I find the code below harder to understand than with immutable.

Reply:
On the GET_DATA_SUCCEEDED, I store the files in `state.files` but `immutablejs` will create a new reference for filters as it is in the same object. In the root of the modal, I fetch the data every params (filter, sort, search, pagination) changes. One of the big advantage with Immer is that there is no new API to learn. You just need to use the produce function